Added support for ZipFile and added more features to IAssembly

cjbush opened the pull request.

jozefizso commented on SystemInterface/Reflection/IAssembly.cs:
```diff
@@ -207,6 +207,7 @@ public interface IAssembly : IEvidenceFactory, ICustomAttributeProvider, ISerial
         /// <param name="assemblyFile">The name or path of the file that contains the manifest of the assembly.</param>
         /// <returns>The loaded assembly. </returns>
         [MethodImpl(MethodImplOptions.NoInlining)]
+		[Obsolete("Use IAssemblyFactory.LoadFrom")]
```
Please fix the indentation.

jozefizso commented on SystemInterface/Reflection/IAssembly.cs:
```diff
@@ -216,6 +217,7 @@ public interface IAssembly : IEvidenceFactory, ICustomAttributeProvider, ISerial
         /// <returns>The loaded assembl.</returns>
         [System.Security.SecuritySafeCritical]
         [MethodImplAttribute(MethodImplOptions.NoInlining)]
+		[Obsolete("Use IAssemblyFactory.Load")]
```
Please fix the indentation.

jozefizso commented on SystemInterface/Reflection/IAssemblyFactory.cs:
```diff
@@ -32,5 +32,15 @@ public interface IAssemblyFactory {
+		/// <returns>The loaded assembly. </returns>
+		[MethodImpl(MethodImplOptions.NoInlining)]
+		IAssembly LoadFrom(string assemblyFile);
+
```
Please fix the whitespace.

jozefizso commented:
Update the code to use 4 space indentation, not tabs.

jozefizso commented:
The files `ZIpFileWrapTests.cs` has wrong CamelCasing.

cjbush commented:
You'll have to check it out and apply your code styling. I'm not resetting all my options in Visual Studio.

jozefizso commented:
These settings does not require you to change your Visual Studio preferences. They can be managed by extensions like Editor Config, ReSharper and others, so Visual Studio uses per-project based code styles.

These are standard C# code style formatting settings. I cannot accept the pull request without it.

cjbush commented:
Not everybody has a Resharper license, you know. I'm pretty sure EditorConfig doesn't support curly braces on the next line, either. I added a Rebracer file, though. That should support everything you want.

And they're *a* standard for C# formatting, not *the* standard. But I've changed everything to match.

jozefizso commented:
Thanks for updating the code style. Rebracer sounds great, I did not want to imply everybody must use ReSharper, just that there are extension to manage per project settings.

jozefizso commented:
Why there is new `AssemblyFactory` class and why were the `IAssembly.LoadFrom` methods marked as obsolete?

cjbush commented:
AssemblyFactory houses all the static methods in the underlying Assembly class, and IAssembly now contains only the instance methods from Assembly (hence the deprecation of LoadFrom, which is a static method in Assembly). Rather than injecting an IAssembly to do a load, you'd inject an IAssemblyFactory. It felt a little clunky to inject an IAssembly, then call Load on that IAssembly which returned another IAssembly. LoadFrom and all of its cousins really are essentially factory methods in the underlying system library, so I think it kind of makes sense to separate those functions out into a separate factory wrapper.

jozefizso commented:
I see, that makes sense for IAssemblyFactory.

I've split the pull request to two parts: one for the Zip implementation and the other will include Assembly related changes. I will merge it and make new release.

cjbush commented:
Fair enough. I wasn't sure if you'd want it as two separate PRs or one. I figured you could split it if you wanted. Thanks!

jozefizso commented:
Implemented in #18 and #19.

I hope did not forget anything when I squashed and split commits.

jozefizso commented:
Thanks for contribution!

cjbush commented:
Absolutely. Thanks for the great library! Sorry about being a curmudgeon about the formatting.
